File: 2ª Unidade/Normalização/Horario.pl
% Relação Horário

horario( [idTurma,horaInicio, horaFim, data]  ).

fds(  [ [[],[idTurma, horaInicio, horaFim, data]] ]  ).

% Query: horario(R), fds(F), threenf(R,F,T).

% Resultado: R == T
F = [[[], [idTurma, horaInicio, horaFim, data]]],
R = [idTurma, horaInicio, horaFim, data],
T = [[data, horaFim, horaInicio, idTurma]]

File: Desenvolvimento/src/entidades/Horario.java
package entidades;

import java.sql.Date;
import java.sql.Time;

public class Horario {
	
	private int idTurma;
	private Time horaInicio;
	private Time horaFim;
	private Date data;
	
	public Horario(int idTurma, Time horaInicio, Time horaFim, Date data) {
		super();
		this.idTurma = idTurma;
		this.horaInicio = horaInicio;
		this.horaFim = horaFim;
		this.data = data;
	}

	public int getIdTurma() {
		return idTurma;
	}

	public void setIdTurma(int idTurma) {
		this.idTurma = idTurma;
	}

	public Time getHoraInicio() {
		return horaInicio;
	}

	public void setHoraInicio(Time horaInicio) {
		this.horaInicio = horaInicio;
	}

	public Time getHoraFim() {
		return horaFim;
	}

	public void setHoraFim(Time horaFim) {
		this.horaFim = horaFim;
	}

	public Date getData() {
		return data;
	}

	public void setData(Date data) {
		this.data = data;
	}
	
}
